lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -256,63 +256,60 @@ def add_level_two_and_three_data(xml, amount, data)
         xml.invoiceReferenceNumber data[:invoice_reference_number] if data.include?(:invoice_reference_number)
         xml.customerReference data[:customer_reference] if data.include?(:customer_reference)
         xml.cardAcceptorTaxId data[:card_acceptor_tax_id] if data.include?(:card_acceptor_tax_id)
-
         {
-          sales_tax: 'salesTax',
+          tax_amount: 'salesTax',
           discount_amount: 'discountAmount',
           shipping_amount: 'shippingAmount',
           duty_amount: 'dutyAmount'
         }.each do |key, tag|
           next unless data.include?(key)
 
           xml.tag! tag do
-            data_amount = data[key].symbolize_keys
-            add_amount(xml, data_amount[:amount].to_i, data_amount)
+            add_amount(xml, data[key].to_i, data)
           end
         end
 
-        xml.discountName data[:discount_name] if data.include?(:discount_name)
-        xml.discountCode data[:discount_code] if data.include?(:discount_code)
+        # xml.discountName data[:discount_name] if data.include?(:discount_name)
+        # xml.discountCode data[:discount_code] if data.include?(:discount_code)
 
-        add_date_element(xml, 'shippingDate', data[:shipping_date]) if data.include?(:shipping_date)
+        # add_date_element(xml, 'shippingDate', data[:shipping_date]) if data.include?(:shipping_date)
 
-        if data.include?(:shipping_courier)
-          xml.shippingCourier(
-            data[:shipping_courier][:priority],
-            data[:shipping_courier][:tracking_number],
-            data[:shipping_courier][:name]
-          )
-        end
+        # if data.include?(:shipping_courier)
+        #   xml.shippingCourier(
+        #     data[:shipping_courier][:priority],
+        #     data[:shipping_courier][:tracking_number],
+        #     data[:shipping_courier][:name]
+        #   )
+        # end
```

Comment:
What is the reason that these are commented out?

Reply:
I was testing to see if these are actually a thing. They're not in WP's docs so I think I'm going to delete them.

Reply:
@aenand were you going to go ahead and remove these?

Reply:
Actually I'm half of the mind to leave them as-is; it wouldn't be the first time an undocumented field was functional and utilized. It's a bit of a gray area, but I'm on board with either.

Reply:
Ah I forgot to. Yeah I'm going to delete them